modules/Model.js

```diff
@@ -1,23 +1,69 @@
 (function(root, factory) {
   if (typeof define === 'function' && define.amd) {
-    define(['underscore', 'backbone', './pollingMixin'], factory);
+    define(['underscore', 'backbone', './pollingMixin', './Events'], factory);
   } else if (typeof exports === 'object') {
-    module.exports = factory(require('underscore'), require('backbone'), require('./pollingMixin'));
+    module.exports = factory(require('underscore'), require('backbone'), require('./pollingMixin'), require('./Events'));
   } else {
     root.Torso = root.Torso || {};
-    root.Torso.Model = factory(root._, root.Backbone, root.Torso.Mixins.polling);
+    root.Torso.Model = factory(root._, root.Backbone, root.Torso.Mixins.polling, root.Torso.Events);
   }
-}(this, function(_, Backbone, pollingMixin) {
+}(this, function(_, Backbone, pollingMixin, Events) {
   'use strict';
 
-  /**
-   * Generic Model
-   * @module    Torso
-   * @class     Model
-   * @constructor
-   * @author kent.willis@vecna.com
-   */
-  var Model = Backbone.Model.extend({});
+
+  var Model = Backbone.Model.extend({
+
+    /**
+     * Overrides constructor to apply timing event triggers to Torso.Events
+     * @method constructor
+     * @override
+     */
+    constructor: function() {
+      Backbone.Model.apply(this,arguments);
+      this.__applyTimingEventListeners(['pre-fetch', 'post-fetch']);
+    },   
+
+    fetch: function(options) {
+      var uuid = _.uniqueId();
```

Comment:
Yea, the thing is though, you can't really trust the uniqueness of something coming from the client side especially if its based on time (concurrent users, etc.). We can make it unique for each session on the browser and have the server generate true unique ids based on time, user information, and the browser session id. Or we could make psuedo-unique numbers on the client.